### PR TITLE
Create a shell script for civet to compile petsc-alt

### LINKED ISSUE #16355: Shell script for installing petsc-alt for civet
## Reason

An old version of PETSc needs to be tested in MOOSE because many users might still use the old PETSc.
Right now, we use conda to load PETSc-alt. It is not controllable via MOOSE PR.
It would be nice if I could control what version of PETSc I need to test.

## Design

Create PETSc-alt script that Civet will call directly. I can control what ingredients  I need to put into this particular version of PETSc by modifying this script.
## Impact
More beautiful code

### PR DESCRIPTION
**Pull request by fdkong**
Closes #16355

### CODE REVIEW

**Comment by fdkong** [comment]
One thing I worry about is that users abuse this script. Anyway to forbidden users from using this script?

**Comment by milljm** [comment]
This is no different than users abusing:

```bash
conda install some-odd-version-of-something
# like:
conda install moose-petsc=3.11.4
```

Or with the older moose-environment package:

```bash
module load moose-dev-clang-alt
# or
module load moose-dev-gcc-alt
```

I wouldn't worry.

**Comment by loganharbour** [comment]
I'll add some meowbuild tests to try this out shortly.

**Comment by loganharbour** [comment]
Can you get it to work with the additional options?

**Comment by fdkong** [comment]
> Can you get it to work with the additional options?

What additional options you want to use?

**Comment by loganharbour** [comment]
https://civet.inl.gov/job/637535/
https://civet.inl.gov/job/637537/
https://civet.inl.gov/job/637536/

Works like a charm with the new recipes.

File: scripts/update_and_rebuild_petsc_alt.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Civet recipe should directly call update_and_rebuild_petsc_alt.sh
```
Comment (loganharbour) [inline scripts/update_and_rebuild_petsc_alt.sh:3]:
```suggestion
# This script is intended for INTERNAL use only by CIVET testing.
#
# It is recommended for all users to use our default version of PETSc,
# which can be compiled using scripts/update_and_rebuild_petsc.sh
#
# If you wish to use an alternate version of PETSc, you should compile
# it on your own and set PETSC_DIR and PETSC_ARCH.
```

File: scripts/update_and_rebuild_petsc_alt.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Give regular users a big warning
+echo "update_and_rebuild_petsc_alt is used for civet only, please do not use it!!!!!"
```
Comment (loganharbour) [inline scripts/update_and_rebuild_petsc_alt.sh:6]:
```suggestion
echo "*** WARNING ***"
echo "scripts/update_and_rebuild_petsc_alt.sh is intended for internal"
echo "use only. Please use scripts/update_and_rebuild_petsc.sh instead."
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — APPROVED** [review]
Merging before test completion - the pending tests are optional tests on the new test boxes.

**Pull request merged**